bigquery/bigquery.go

```diff
@@ -173,24 +173,40 @@ func (c *Client) insertJob(ctx context.Context, job *bq.Job, media io.Reader) (*
 // Due to differences in options it supports, it cannot be used for all existing
 // jobs.insert requests that are query jobs.
 func (c *Client) runQuery(ctx context.Context, queryRequest *bq.QueryRequest) (*bq.QueryResponse, error) {
-	call := c.bqs.Jobs.Query(c.projectID, queryRequest)
-	setClientHeader(call.Header())
-
-	var res *bq.QueryResponse
-	var err error
-	invoke := func() error {
-		sCtx := trace.StartSpan(ctx, "bigquery.jobs.query")
-		res, err = call.Do()
-		trace.EndSpan(sCtx, err)
-		return err
+	type respS struct {
+		res *bq.QueryResponse
+		err error
 	}
+	ch := make(chan *respS, 1)
+	go func() {
```

Comment:
if we pass the context to the call obj, can all of this extra code be avoided?

Reply:
Good spot, thanks for calling that out.  removed the outer context patterns and just plumbed the context into the discovery rpcs.